### PR TITLE
Inline RSpec/InstanceVariable todos

### PR DESCRIPTION
**Pull request by dduugg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

As a follow-up to https://github.com/Homebrew/brew/pull/14400 this inlines the disables for `RSpec/InstanceVariable`, allowing us to remove one of the `.rubocop_todo.yml` files. 

I considered some fixes instead (for some of the git specs, one could instead use the git reflog rather than recording shas in ivars, although that is subtly different test), but ultimately concluded these were reasonable enough to just disable the violations for now.

### CODE REVIEW

**Review by carlocab — APPROVED** [review]
Looks reasonable to me. Thanks @dduugg!

File: Library/Homebrew/test/download_strategies/git_spec.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
   def git_commit_all
     system "git", "add", "--all"
-    system "git", "commit", "-m", "commit number #{@commit_id}"
+    system "git", "commit", "-m", "commit number #{@commit_id}" # rubocop:disable RSpec/InstanceVariable
     @commit_id += 1
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/download_strategies/git_spec.rb:22]:
```suggestion
    # Allow instance variables here to have nice commit messages.
    # rubocop:disable RSpec/InstanceVariable
    system "git", "commit", "-m", "commit number #{@commit_id}"
    @commit_id += 1
    # rubocop:enable RSpec/InstanceVariable
```

File: Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb [hunk 2]
```diff
@@ -119,7 +119,7 @@ def brew(*args)
     end
 
     Bundler.with_unbundled_env do
-      stdout, stderr, status = Open3.capture3(env, *@ruby_args, *args)
+      stdout, stderr, status = Open3.capture3(env, *@ruby_args, *args) # rubocop:disable RSpec/InstanceVariable
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb:122]:
```suggestion
      # Allow instance variable here to improve performance through memoization.
      # rubocop:disable RSpec/InstanceVariable
      stdout, stderr, status = Open3.capture3(env, *@ruby_args, *args)
      # rubocop:enable RSpec/InstanceVariable
```

File: Library/Homebrew/test/utils/git_spec.rb [hunk 3]
```diff
@@ -45,12 +45,14 @@
   end
 
   let(:file) { "README.md" }
+  # rubocop:disable RSpec/InstanceVariable
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/utils/git_spec.rb:48]:
```suggestion
  let(:file) { "README.md" }
  # Allow instance variables here for a simpler `before do` block.
  # rubocop:disable RSpec/InstanceVariable
```

**Review by MikeMcQuaid — APPROVED** [review]
Great work here again @dduugg! Added a couple of comments here. For future reference we're aiming to have these whenever we have disables.

**Comment by dduugg** [comment]
@MikeMcQuaid Ack, will note going forward. Thanks for fixing in this PR.

**Pull request merged**